### PR TITLE
added negative value support in pct_changes of generic.py

### LINKED ISSUE #56618: BUG: pct_change not calculated correctly (wrong sign)
### Pandas version checks

- [X] I have checked that this issue has not already been reported.

- [X] I have confirmed this bug exists on the [latest version](https://pandas.pydata.org/docs/whatsnew/index.html) of pandas.

- [X] I have confirmed this bug exists on the [main branch](https://pandas.pydata.org/docs/dev/getting_started/install.html#installing-the-development-version-of-pandas) of pandas.

### Reproducible Example

```python
import pandas as pd
test = pd.DataFrame()
test['testdata'] = [-68, 5144]
test['pct_changes'] = test['testdata'].pct_change()
print (test['pct_changes'])
0     0.000000
1   -76.647059
Name: pct_changes, dtype: float64
```

### Issue Description

correct result should be 76.647059 and not -76.647059. It seems the current implementation of pct_change() lacks the abs() function and thus returns the wrong sign

### Expected Behavior

percentage_change = ((new_value - old_value) / abs(old_value)) * 100
((5144 - (-68)) / abs(-68)) * 100 = 76.647059

### Installed Versions

<details>

INSTALLED VERSIONS
------------------
commit              : a671b5a8bf5dd13fb19f0e88edc679bc9e15c673
python              : 3.11.6.final.0
python-bits         : 64
OS                  : Linux
OS-release          : 5.15.133.1-microsoft-standard-WSL2
Version             : #1 SMP Thu Oct 5 21:02:42 UTC 2023
machine             : x86_64
processor           : x86_64
byteorder           : little
LC_ALL              : None
LANG                : en_US.UTF-8
LOCALE              : en_US.UTF-8

pandas              : 2.1.4
numpy               : 1.24.3
pytz                : 2023.3.post1
dateutil            : 2.8.2
setuptools          : 68.2.2
pip                 : 23.3.1
Cython              : None
pytest              : None
hypothesis          : None
sphinx              : None
blosc               : None
feather             : None
xlsxwriter          : None
lxml.etree          : 4.9.3
html5lib            : 1.1
pymysql             : None
psycopg2            : None
jinja2              : 3.1.2
IPython             : 8.17.2
pandas_datareader   : 0.10.0
bs4                 : 4.12.2
bottleneck          : None
dataframe-api-compat: None
fastparquet         : None
fsspec              : 2023.9.1
gcsfs               : None
matplotlib          : 3.8.0
numba               : None
numexpr             : None
odfpy               : None
openpyxl            : None
pandas_gbq          : None
pyarrow             : 13.0.0
pyreadstat          : None
pyxlsb              : None
s3fs                : None
scipy               : 1.11.2
sqlalchemy          : None
tables              : None
tabulate            : None
xarray              : None
xlrd                : 2.0.1
zstandard           : None
tzdata              : 2023.3
qtpy                : None
pyqt5               : None

</details>

### PR DESCRIPTION
**Pull request by paridhisharma24**
closes #56618

### CODE REVIEW

**Comment by asishm-wk** [comment]
Thanks for the PR. Few points:

1. We can't change the default behavior without a deprecation cycle here (and reading the comments of the issue, don't think the default needs to change either).
2. You'll need to add another argument/parameter to the `pct_change` method that indicates whether or not you want to take absolute value of the denominator (similar changes would be required in your tests as well)

**Comment by paridhisharma24** [comment]
@asishm-wk I am not able to figure out the issue with the failing check
Could you please help with the same

**Comment by asishm** [comment]
The error comes from mypy - if you click on the failed build.

> pandas/core/generic.py:12152: error: Unsupported left operand type for / ("Self")  [operator]
> pandas/core/generic.py:12152: note: Right operand is of type "Self | DataFrame"

Looks like you removed the type ignore comment when you made the changes (needs to be there in both the if and else statements).

File: pandas/core/generic.py [hunk 1]
```diff
@@ -12145,7 +12146,10 @@ def pct_change(
 
         shifted = data.shift(periods=periods, freq=freq, axis=axis, **kwargs)
         # Unsupported left operand type for / ("Self")
-        rs = data / shifted - 1  # type: ignore[operator]
+        if use_absolute_value:
+            rs = rs = (data - shifted) / abs(shifted)  # type: ignore[operator]
```
Comment (asishm) [inline pandas/core/generic.py:12150]:
```suggestion
            rs = (data - shifted) / abs(shifted)  # type: ignore[operator]
```

File: pandas/tests/dtypes/test_generic.py [hunk 2]
```diff
@@ -1,3 +1,4 @@
+import random
```
Comment (asishm) [inline pandas/tests/dtypes/test_generic.py:1]: is the import and setting the seed below required?

**Review by asishm — COMMENTED** [review]
tests should likely go in `pandas/tests/series/methods/test_pct_change.py` and see if you can  use an existing test and parametrize the new argument.

File: pandas/tests/dtypes/test_generic.py (hunk 2 above)
Comment (paridhisharma24) [reply pandas/tests/dtypes/test_generic.py:1]: will check and remove if not required

**Comment by WillAyd** [comment]
I appreciate the PR but I am not sure it is worth expanding the API for this.

**Comment by mroeschke** [comment]
Thanks for the PR but it is a bit too early to craft a solution to the issue as it's still being actively discussed. Going to close but we can reopen if it has more acceptance from the core developers

**Pull request closed without merging**